### PR TITLE
Adding ability to change font size in PDF form fields

### PR DESCRIPTION
**Pull request by duffyd**
#727 Here's the PR. Please let me know if there's anything that needs to be changed. Would be great if this could be released soon so we can start using it in our app 🙂

### CODE REVIEW

File: apps/node/tests/test19.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { Assets } from '..';
```
Comment (Hopding) [inline apps/node/tests/test19.ts:1]: This is a good test, but please add the logic to one of the existing form filling tests. The integration tests take a lot of time to run, so we try to keep the number of files to a minimum.

Comment (Hopding) [reply apps/node/tests/test19.ts:1]: Also, please be sure to add the same logic to the `deno`, `web`, and `rn` tests. The tests should always be identical across each runtime/environment. Note that you may not be able to actually run the `rn` tests. That's fine, but please still update the file. I'll run it along with the others before releasing the changes.

File: src/core/acroform/PDFAcroField.ts [hunk 2]
```diff
@@ -7,6 +7,20 @@ import PDFNumber from 'src/core/objects/PDFNumber';
 import PDFArray from 'src/core/objects/PDFArray';
 import PDFRef from 'src/core/objects/PDFRef';
 
+const findLastMatch = (value: string, regex: RegExp) => {
```
Comment (Hopding) [inline src/core/acroform/PDFAcroField.ts:10]: This function is very similar to the following: https://github.com/Hopding/pdf-lib/blob/4be96eb269910bbc6d3ddcd48f2a6382742e179b/src/api/form/appearances.ts#L111-L121

Please move this function into https://github.com/Hopding/pdf-lib/blob/master/src/utils/strings.ts and remove the function defined in `appearances.ts` to avoid duplicate code and logic.

File: src/core/acroform/PDFAcroField.ts [hunk 3]
```diff
@@ -7,6 +7,20 @@ import PDFNumber from 'src/core/objects/PDFNumber';
...
+  while (position < value.length) {
+    const match = value.substring(position).match(regex);
+    if (!match) return { match: lastMatch, pos: position };
+    lastMatch = match;
+    position += (match.index ?? 0) + match[0].length;
+  }
+  return { match: lastMatch, pos: position };
+};
+
+const tfRegex = /\/([^\0\t\n\f\r\ ]+)[\0\t\n\f\r\ ]*(\d*\.\d+|\d+)?[\0\t\n\f\r\ ]+Tf/;
```
Comment (Hopding) [inline src/core/acroform/PDFAcroField.ts:22]: Please add some comments illustrating how this regex matches various inputs. Similar to what was done here: https://github.com/Hopding/pdf-lib/blob/4be96eb269910bbc6d3ddcd48f2a6382742e179b/src/api/form/appearances.ts#L123-L126

File: src/core/acroform/PDFAcroField.ts [hunk 4]
```diff
@@ -96,6 +110,24 @@ class PDFAcroField {
     return DA?.asString();
   }
 
+  setFontSize(fontSize: number) {
+    if (this.DA()) {
+      const da = this.getDefaultAppearance()!;
+      const daMatch = findLastMatch(da, tfRegex);
```
Comment (Hopding) [inline src/core/acroform/PDFAcroField.ts:116]: Can we do the following to make the null check safer and avoid the non-null assertion?
```suggestion
  setFontSize(fontSize: number) {
    const da = this.getDefaultAppearance();
    if (da) {
      const daMatch = findLastMatch(da, tfRegex);
```

File: src/core/acroform/PDFAcroField.ts [hunk 5]
```diff
@@ -96,6 +110,24 @@ class PDFAcroField {
     return DA?.asString();
   }
 
+  setFontSize(fontSize: number) {
```
Comment (Hopding) [inline src/core/acroform/PDFAcroField.ts:113]: Does this function work for newly created text fields? E.g. https://pdf-lib.js.org/docs/api/classes/pdfform#createtextfield. It's important that form methods work for all types of fields.

Comment (Hopding) [inline src/core/acroform/PDFAcroField.ts:113]: Please add wrapper methods to [`PDFTextField`](https://github.com/Hopding/pdf-lib/blob/master/src/api/form/PDFTextField.ts), [`PDFButton`](https://github.com/Hopding/pdf-lib/blob/master/src/api/form/PDFButton.ts), [`PDFDropdown`](https://github.com/Hopding/pdf-lib/blob/master/src/api/form/PDFDropdown.ts), and [`PDFOptionList`](https://github.com/Hopding/pdf-lib/blob/master/src/api/form/PDFOptionList.ts) with the appropriate doc comments. Defining the logic in the low-level `PDFAcroField` API makes sense, but this is functionality that should also be available in the high-level API. Most users should not need to access the `.acroField` property or the low-level core APIs.

Note that the wrapper methods should assert that their `fontSize` parameter is non-negative.

Comment (Hopding) [inline src/core/acroform/PDFAcroField.ts:113]: Please add some unit tests for this method to https://github.com/Hopding/pdf-lib/blob/master/tests/core/acroform/PDFAcroField.spec.ts. Parsing logic like this is error prone so it's important to test carefully to avoid regressions and missed edge cases.

**Review by Hopding — CHANGES_REQUESTED** [review]
Thanks for working on this @duffyd! Your approach looks solid and the code is good. I've requested some changes. Once these things are taken care of this can be merged and released.

File: src/core/acroform/PDFAcroField.ts (hunk 5 above)
Comment (duffyd) [reply src/core/acroform/PDFAcroField.ts:106]: Yes, I've also implemented integration tests for this. The node one is here: https://github.com/duffyd/pdf-lib/blob/efbc0ee4da21d7cb6a4ed4f61859326148449e8b/apps/node/tests/test17.ts#L110-L113

**Comment by duffyd** [comment]
@Hopgding I've implemented all requested changes and they're in my fork now. Could you please review and let me know if you're happy?

**Comment by Hopding** [comment]
Thanks @duffyd! I'll take a look as soon as I get a chance. Looks like there's a conflict though. Would you please take a look at that?

**Comment by duffyd** [comment]
> Thanks @duffyd! I'll take a look as soon as I get a chance. Looks like there's a conflict though. Would you please take a look at that?

I've fixed the conflict but it's still complaining. Sorry I'm not a git expert. Are you able to resolve the conflict?

**Comment by duffyd** [comment]
Ok, after *reading the documentation* I have now properly resolved the conflict.

**Review by Hopding — APPROVED** [review]
Looks great @duffyd! Really appreciate your work on this. It'll go out in the next release of pdf-lib (hopefully tonight or tomorrow).

**Comment by duffyd** [comment]
Thanks!!
